.ci/containers/membership-checker/community.go

```diff
@@ -26,6 +26,46 @@ func approveCommunityChecker(prNumber, projectId, commitSha string) error {
 	return nil
 }
 
+func postAwaitingApprovalBuildLink(prNumber, GITHUB_TOKEN, projectId, commitSha string) error {
+	buildId, err := getPendingBuildId(projectId, commitSha)
+	if err != nil {
+		return err
+	}
+
+	if buildId == "" {
+		return fmt.Errorf("Failed to find pending build for PR %s", prNumber)
+	}
+
+	targetUrl := fmt.Sprintf("https://console.cloud.google.com/cloud-build/builds;region=global/%s?project=%s", buildId, projectId)
+
+	postBody := map[string]string{
+		"context":    "Approve Build",
+		"state":      "success",
```

Comment:
I was also hesitating about it a bit. Yeah, if we set this to be pending, we'll have to flip it. We can only have the flip step for community PRs, as auto-run PRs don't have this status. But for users who join the membership after they create a PR, this check will always remain pending as there's no way we can change it I assume (my understanding is that if we don't remove the status, it will not be updated and will always show on GH UI).
Plus, I feel like we can consider it more like a button instead of a status, and we still use the state of `changelog-checker` as a indicator.

Reply:
If on auto run we just post the approval in the generate diffs run irregardless of the presence of pending it should be fine no?

Reply:
Then, basically we're gonna have `Approve Build` status posted for every PR, which I'm not sure if it's confusing. Plus, if we add the part that posts the approval in the generate diffs, then we need to either add an extra step (slow down the generate diffs) or add the feature in one of the current steps,`github-differ` likely (add more time to generate diffs + rewrite some of the current code to bash). Also we're gonna have the source code of this feature in multiple places and I'm not sure if that's ideal.

I guess we can rename this status to something like `community-checker link` or similar to indicate that the status is not supposed to tell if the build is waiting for approval but just a temporary workaround to post build link before cloud build post it directly. Success state of this status just means the build is posted successfully.

Reply:
As discussed offline, I'll play around with the approaches for implementing pending/approval for the status and will possibly add it in another PR